### PR TITLE
Update GHC to 9.2.4 and dependencies

### PR DESCRIPTION
**Pull request by arcz**
## Description
There was a big merge of haskell-updates branch to master in nixpkgs. The most visible changes are `brick` and `aeson` that got significantly bumped.

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

File: hevm.cabal [hunk 1]
```diff
@@ -130,35 +112,35 @@ library
     Decimal                           >= 0.5.1 && < 0.6,
     containers                        >= 0.6.0 && < 0.7,
     deepseq                           >= 1.4.4 && < 1.5,
-    time                              >= 1.8.0 && < 1.11,
+    time,
```
Comment (d-xo) [inline hevm.cabal:115]: Can we add version bounds to all the deps here. Hackage are kinda fussy about it.

File: src/EVM/TTY.hs [hunk 2]
```diff
@@ -317,142 +317,165 @@ takeStep
...
-  n ->
-    let
-      (step, (vm, stepper)) = fromJust $ lookupLT n (view uiSnapshots s)
-      s1 = s
-        & set uiVm vm
-        & set (uiVm . cache) (view (uiVm . cache) s)
-        & set uiStep step
-        & set uiStepper stepper
-      stepsToTake = n - step - 1
+  => UiVmState -> IO UiVmState -- EventM n UiState ()
```
Comment (d-xo) [inline src/EVM/TTY.hs:369]: can we remove the comment?

File: src/EVM/TTY.hs [hunk 3]
```diff
@@ -317,142 +317,165 @@ takeStep
...
-    browserContractList
-    handleListEvent
-    e
-  continue (ViewContracts s')
+appEvent (VtyEvent e@(V.EvKey V.KDown [])) = get >>= \case
+  ViewContracts _s -> do
+    Brick.zoom
+      (_ViewContracts . browserContractList)
+      (handleListEvent e)
+    pure () -- (ViewContracts s')
```
Comment (d-xo) [inline src/EVM/TTY.hs:405]:
```suggestion
    pure ()
```

File: src/EVM/TTY.hs (hunk 2 above)
Comment (d-xo) [inline src/EVM/TTY.hs:369]:
```suggestion
  => UiVmState -> IO UiVmState
```

**Comment by d-xo** [comment]
can we rebase on `main` and fix any outstanding compiler warnings pls :)

File: hevm.cabal [hunk 4]
```diff
@@ -186,7 +168,7 @@ library
     smt2-parser                       >= 0.1.0.1,
     word-wrap                         >= 0.5 && < 0.6,
     spool                             >= 0.1 && < 0.2,
-    parsec                            >= 3.1.14.0 && < 3.1.15
+    parsec
```
Comment (d-xo) [inline hevm.cabal:171]: why did we remove the bound here?

File: src/EVM/TTY.hs [hunk 5]
```diff
@@ -317,142 +317,165 @@ takeStep
...
-    e
-  continue (ViewContracts s')
+appEvent (VtyEvent e@(V.EvKey V.KDown [])) = get >>= \case
+  ViewContracts _s -> do
+    Brick.zoom
+      (_ViewContracts . browserContractList)
+      (handleListEvent e)
+    pure ()
+  ViewVm s ->
+    if view uiShowMemory s then
```
Comment (d-xo) [inline src/EVM/TTY.hs:407]: I think we can just ignore the `down` and `up` keys for the memory view. It's already scrollable with the scrollwheel

File: hevm.cabal (hunk 4 above)
Comment (arcz) [reply hevm.cabal]: not sure but it's fixed

File: src/EVM/TTY.hs (hunk 5 above)
Comment (arcz) [reply src/EVM/TTY.hs]: done

**Review by d-xo — APPROVED** [review]
Thanks!

**Pull request merged**